### PR TITLE
Fix spelling errors in dartdocs

### PR DESCRIPTION
**Pull request by gspencergoog**
Fix the spelling errors in the dartdocs for the engine.

There are no functionality changes here, just documentation fixes.

### CODE REVIEW

File: lib/stub_ui/painting.dart [hunk 1]
```diff
@@ -1762,17 +1762,17 @@ class EngineLayer {
 
 /// A complex, one-dimensional subset of a plane.
 ///
-/// A path consists of a number of subpaths, and a _current point_.
+/// A path consists of a number of sub-paths, and a _current point_.
```
Comment (cbracken) [inline lib/stub_ui/painting.dart]: We should check the Skia docs on this one and be consistent with them. Subjectively, this looks off to me.

File: lib/stub_ui/painting.dart [hunk 2]
```diff
@@ -59,7 +59,7 @@ Color _scaleAlpha(Color a, double factor) {
 ///
 /// Consider the light teal of the Flutter logo. It is fully opaque, with a red
 /// channel value of 0x42 (66), a green channel value of 0xA5 (165), and a blue
-/// channel value of 0xF5 (245). In the common "hash syntax" for colour values,
+/// channel value of 0xF5 (245). In the common "hash syntax" for color values,
```
Comment (cbracken) [inline lib/stub_ui/painting.dart]: Based on where this is I'm going to point the finger at @Hixie. :)

File: lib/stub_ui/compositing.dart [hunk 3]
```diff
@@ -215,7 +215,7 @@ class SceneBuilder {
   /// Android view: When resizing an Android view there is a short period during
   /// which the framework cannot tell if the newest texture frame has the
   /// previous or new size, to workaround this the framework "freezes" the
-  /// texture just before resizing the Android view and unfreezes it when it is
+  /// texture just before resizing the Android view and un-freezes it when it is
```
Comment (cbracken) [inline lib/stub_ui/compositing.dart]: I thought US English preferred unhyphenated variants (e.g. Commonwealth 'co-ordinate', vs US 'coordinate'). 

The unhyphenated version looks more natural to me and I suspect we'll see this re-introduced (see what I did there) if we don't make this sort of thing a presubmit.

File: lib/stub_ui/painting.dart (hunk 1 above)
Comment (cbracken) [reply lib/stub_ui/painting.dart]: Objectively, I can't find a single instance of 'subpath' on Skia.org, but I do find one instance of 'sub-path'. Objection overruled. Proceed, sir.

**Comment by cbracken** [comment]
Couple of nits, but overall LGTM.

File: lib/stub_ui/compositing.dart (hunk 3 above)
Comment (gspencergoog) [reply lib/stub_ui/compositing.dart]: My understanding of US English hyphenation is that we use it when a term is less common, but we do use it when the unhyphenated version (see what I did there? FYI, Chrome flagged unhyphenated as a spelling error...) becomes mainstream, and a word unto itself.  In this case "unfreeze" didn't seem common enough to me, but I could totally be persuaded to not hyphenate: it was just a subjective judgement call.

**Pull request merged**